src/queue.c

```diff
@@ -0,0 +1,83 @@
+#include "queue.h"
+
+IG_Queue * IGQueue_new(){
+
+	// Init queue
+	IG_Queue* queue = (IQ_Queue*)malloc(sizeof(IG_Queue));
+	queue->size = 0;
+	queue->front = NULL;
+	queue->end = NULL;
+
+	// Init mutex for this specific queue
+	pthread_mutex_init(queue->mutex,NULL);
+	return queue;
+}
+
+// Function to enqueue data
+void IG_Queue_put(IG_Queue* queue, IG_Data* new_data){
+
+	// Trying to lock this queue from other threads, if it's already locked this thread will
+	// get locked and will be blocked until the mutex is unlocked by another thread
+	pthread_mutex_lock(queue->mutex);
+
+	// Creating new element for the queue
+	IG_QElement* element = (IG_QElement*)malloc(sizeof(IG_QElement));
+	
+	// TODO: Sollen die Daten einfach der pointer weiter ggb. werden oder neu im HEAP?
+	// Mit pointer weniger aufwand wegen kopieren der pointer and weniger speicherverbrauch 	
```

Comment:
würde das in der funktion so lassen --
wenn wir ne funktion mit kopieren brauchen machen wir das in ner extra funktion
kannst aber noch nen kommentar in der .h hinzufügen der dadrauf hinweißt